include/sys/spa.h

```diff
@@ -408,6 +488,26 @@ _NOTE(CONSTCOND) } while (0)
 #define	BP_GET_LEVEL(bp)		BF64_GET((bp)->blk_prop, 56, 5)
 #define	BP_SET_LEVEL(bp, x)		BF64_SET((bp)->blk_prop, 56, 5, x)
 
+/* encrypted, authenticated, and MAC cksum bps use the same bit */
+#define	BP_USES_CRYPT(bp)		BF64_GET((bp)->blk_prop, 61, 1)
+#define	BP_SET_CRYPT(bp, x)		BF64_SET((bp)->blk_prop, 61, 1, x)
+
+#define	BP_IS_ENCRYPTED(bp)			\
+	(BP_USES_CRYPT(bp) &&			\
+	BP_GET_LEVEL(bp) <= 0 &&		\
+	DMU_OT_IS_ENCRYPTED(BP_GET_TYPE(bp)))
```

Comment:
> My biggest concern there is that we would be losing redundancy unnecessarily. Currently, if you set copies=2, ZFS will give you 3 copies of your metadata. With the current implementation, any data that is authenticated can still have all 3 copies.

That's true but I'm not sure it's important.  We're already losing the 3rd copy of dnode blocks, right (since the bonus buffers are encrypted)?  This is talking about losing the 3rd copy of a handful of objects below that (master node, SA metadata).